tests/pytest_tests/unit_tests/test_launch/test_runner/test_kubernetes.py

```diff
@@ -0,0 +1,52 @@
+import pytest
+
+from wandb.sdk.launch.runner.kubernetes_runner import add_label_to_pods, add_wandb_env
+
+
+@pytest.fixture
+def manifest():
+    return {
+        "kind": "Job",
+        "spec": {
+            "template": {
+                "spec": {
+                    "containers": [
+                        {
+                            "name": "master",
+                            "image": "${image_uri}",
+                            "imagePullPolicy": "IfNotPresent",
+                        },
+                        {
+                            "name": "worker",
+                            "image": "${image_uri}",
+                            "workingDir": "/home",
+                            "imagePullPolicy": "IfNotPresent",
+                        },
+                    ],
```

Comment:
These tests are sweet. I'd recommend adding `"env": {"name": "TEST_EXISTING_ENV", "value": "test_existing_value"}` to one of the containers, and assert in `test_add_env` that the existing env isn't overwritten by add_wandb_env

Reply:
Mm, yeah the behavior here is to extend the list of environment variable definitions so there would just be duplicate keys in that list.

Reply:
I'm not sure how k8s handles that. It might cause an error. But I think it's actually good to override variables because we are only dealing with WANDB_ variables here and I think the agent should be setting all of those.

Reply:
Chat gpt:

When you include multiple values for the same environment variable in a Kubernetes container spec, the last defined value takes precedence, and that's the value the container will see and use. This is because, during the environment variable evaluation process, the environment variables are processed in the order they appear in the container spec, and duplicates are overwritten.

Here's an example:

```yaml
apiVersion: v1
kind: Pod
metadata:
  name: example-pod
spec:
  containers:
  - name: example-container
    image: example-image
    env:
    - name: MY_VARIABLE
      value: "first-value"
    - name: MY_VARIABLE
      value: "second-value"
```

In this example, the `MY_VARIABLE` environment variable is set to "first-value" and then to "second-value". The container will only see the value "second-value" for `MY_VARIABLE`, as it is the last value defined for that environment variable.

It's always a good practice to avoid such duplicate environment variable definitions in your Kubernetes container spec to prevent confusion and potential errors.

Reply:
Yeah sorry to clarify, I meant a case where the container has `ENV_1` set, and the function is called, setting `ENV_2` and `ENV_3`. Then assert that all three are set.

I agree with you about overwriting the same env var.

Reply:
Ok got it, yeah I will add that check